prophet-ressa-minify/res/deathstarbench/callgraph/ressa.json

```diff
@@ -0,0 +1,90 @@
+[
+    {
+      "identifier": "Field",
+      "pattern": "#{pool_name}",
+      "auxiliary_pattern": "ClientPool<ThriftClient<#{service_name}ServiceClient>>",
+      "subpatterns": [],
+      "callback": "service_and_pool_name.rn",
+      "essential": true
+    },
+    {
+      "identifier": "DeclStmt",
+      "pattern": "",
+      "subpatterns": [
+        {
+          "identifier": "VarDecl",
+          "pattern": "#{wrapper_name}(.*_client_wrapper)",
+          "subpatterns": [],
+          "essential": true
+        },
+        {
+          "identifier": "CallExpr",
+          "pattern": "Pop",
+          "auxiliary_pattern": "#&{pool_name}(.*_client_pool)",
+          "subpatterns": [],
+          "essential": false
+        }
+      ],
+      "callback": "wrapper_and_pool_name.rn",
+      "essential": true
+    },
+    {
+      "identifier": "DeclStmt",
+      "pattern": "",
+      "subpatterns": [
+        {
+          "identifier": "VarDecl",
+          "pattern": "#{client_name}(.*_client)",
+          "subpatterns": [],
+          "essential": true
+        },
+        {
+          "identifier": "CallExpr",
+          "pattern": "GetClient",
+          "auxiliary_pattern": "#&{wrapper_name}",
+          "subpatterns": [],
+          "essential": true
+        }
+      ],
+      "callback": "client_and_wrapper_name.rn",
+      "essential": true
+    },
+    {
+      "identifier": "CallExpr",
+      "pattern": "#{endpoint_name}",
+      "auxiliary_pattern": "#&{client_name}(.*_client$)",
+      "subpatterns": [],
+      "callback": "client_and_endpoint_name.rn",
+      "essential": true
+    },
+    {
+      "identifier": "ClassOrInterface",
+      "pattern": "#{callee_name}",
+      "subpatterns": [
+        {
+          "identifier": "CallExpr",
+          "pattern": "#&{endpoint_name}",
+          "auxiliary_pattern": "#&{client_name}(.*_client$)",
+          "subpatterns": [],
+          "essential": true,
+          "callback": "callee_and_endpoint_name.rn"
+        }
+      ],
+      "essential": true
+    },
+    {
+      "identifier": "ClassOrInterface",
+      "pattern": "#{callee_name}",
+      "subpatterns": [
+        {
+          "identifier": "Field",
+          "pattern": "#{entity_attribute}",
+          "auxiliary_pattern": "#{attribute_type}",
+          "subpatterns": [],
+          "essential": true,
+          "callback": "final.rn"
```

Comment:
I'm not sure what's up with this ReSSA parser. Its script (prophet-ressa-minify/res/deathstarbench/callgraph/final.rn) doesn't match the attributes it scraped. Are you sure there wasn't a copy error?

Reply:
https://github.com/cloudhubs/source-code-parser/blob/main/ressa_examples/deathstarbench_ressa_endpoint.json

The last 2 callbacks in this file actually point to the same one so I just de-duplicated them, but I believe this is exactly what was in the original ReSSA

Reply:
I don't think this is a copy error, so the ReSSA itself may actually be out of date if this isn't correct

Reply:
Interesting. Yeah, that parser does nothing, just tried running it without it. Should we delete it?

Reply:
Yeah, I think it can be removed. It looks like it's actually something that might be related to mongo parsing since it's referencing `entity_attribute` etc